src/core_plugins/elasticsearch/lib/upgrade_config.js

```diff
@@ -32,25 +33,22 @@ export default function (server) {
 
     // Look for upgradeable configs. If none of them are upgradeable
     // then create a new one.
-    const body = _.find(response.hits.hits, isUpgradeable.bind(null, server));
-    if (!body) {
+    const configSavedObject = _.find(configSavedObjects, isUpgradeable.bind(null, server));
+    if (!configSavedObject) {
       return createNewConfig();
     }
 
     // if the build number is still the template string (which it wil be in development)
     // then we need to set it to the max interger. Otherwise we will set it to the build num
-    body._source.buildNum = config.get('pkg.buildNum');
+    configSavedObject.attributes.buildNum = config.get('pkg.buildNum');
```

Comment:
Good point - that is on the client-side. Sometime in the future I should move the SavedObject to common and use for both client and server-side.